src/grow_worker/worker.py

```diff
@@ -67,36 +70,40 @@ class CalculationResult:
     output_esdl: Optional[str]
 
 
-def run_rtc_calculation(job_id: uuid4, esdl_string: str, workflow_type: WorkFlowType) -> CalculationResult:
-    input_file = Path(os.environ["INPUT_FILES_DIR"]) / f"{job_id}.esdl"
-    output_file = Path(os.environ["OUTPUT_FILES_DIR"]) / f"{job_id}.esdl"
+def rtc_calculate(job_id: uuid4, esdl_string: str, workflow_type: WorkFlowType, update_progress: Callable) -> Any:
```

Comment:
Missing types of Callable for update_progress. Should add arguments & return type.